package-res/resources/web/pentaho/type/complex.js

```diff
@@ -608,6 +608,46 @@ define([
       //endregion
       //endregion
 
+      //region serialization
+      serializeComplexValues: function(spec, values, keyArgs){
+        for (var property in values) {
```

Comment:
Iterating objects' keys is slow by nature. More so if over local and inherited properties, as this construct does. It's much better to iterate arrays when possible, and we can. Check how the `Complex` constructor does it: https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/package-res/resources/web/pentaho/type/complex.js#L102

But the constructor is a performance bottleneck. You could use the Complex.Meta#each method, here, instead of accessing _getProps() directly.